### PR TITLE
[AIRFLOW-1917] Remove extra newline character from log

### PR DESCRIPTION
**Pull request by ckljohn**
Make sure you have checked _all_ steps below.

### Jira

- [x] My PR addresses the following [Airflow Jira](https://issues.apache.org/jira/browse/AIRFLOW/) issues and references them in the PR title. For example, "\[AIRFLOW-XXX\] My Airflow PR"
  - https://issues.apache.org/jira/browse/AIRFLOW-1917
  - In case you are fixing a typo in the documentation you can prepend your commit with \[AIRFLOW-XXX\], code changes always need a Jira issue.

### Description

- [x] Here are some details about my PR, including screenshots of any UI changes:
Remove extra newline character from log.

### Tests

- [X] My PR adds the following unit tests __OR__ does not need testing for this extremely good reason:
Include a unit test change.

### Commits

- [x] My commits all reference Jira issues in their subject lines, and I have squashed multiple commits if they address the same issue. In addition, my commits follow the guidelines from "[How to write a good git commit message](http://chris.beams.io/posts/git-commit/)":
  1. Subject is separated from body by a blank line
  1. Subject is limited to 50 characters (not including Jira issue reference)
  1. Subject does not end with a period
  1. Subject uses the imperative mood ("add", not "adding")
  1. Body wraps at 72 characters
  1. Body explains "what" and "why", not "how"

### Documentation

- [ ] In case of new functionality, my PR adds documentation that describes how to use it.
  - When adding new operators/hooks/sensors, the autoclass documentation generation needs to be added.

### Code Quality

- [x] Passes `git diff upstream/master -u -- "*.py" | flake8 --diff`

### CODE REVIEW

**Comment by r39132** [comment]
@ckljohn Any idea where this extra newline is being created? I'd rather remove the source of the extra newline than strip it afterwards. This feels like a hack.

**Comment by ashb** [comment]
The extra line comes from reading stdout/stderr of the sub-process, so it's unavoidable - check out the linked jira for a repro case involving an operator. this StreamLogWriter is used to capture StdOut from operators etc.

**Comment by r39132** [comment]
@ashb Not sure I grok your comment fully? Are you giving this a +1?

**Comment by ckljohn** [comment]
It basically follows how `base_task_runner.py` handle stream logs.
https://github.com/apache/incubator-airflow/blob/48229328f28bfd1717006c5346ba64985a97935d/airflow/task/task_runner/base_task_runner.py#L98-L107

File: airflow/utils/log/logging_mixin.py [hunk 1]
```diff
@@ -92,7 +92,7 @@ def write(self, message):
             self._buffer += message
         else:
             self._buffer += message
-            self.logger.log(self.level, self._buffer)
+            self.logger.log(self.level, self._buffer.rstrip('\n'))
```
Comment (Fokko) [inline airflow/utils/log/logging_mixin.py]: @r39132 I think you will get this `\n` for free, therefore I think this is a good change. Wouldn't `.rstrip()` be sufficient, to also trim all kinds of whitespace at the end.
```
MacBook-Pro-van-Fokko:~ fokkodriesprong$ python3
Python 3.7.0 (default, Jun 29 2018, 20:13:13) 
[Clang 9.1.0 (clang-902.0.39.2)] on darwin
Type "help", "copyright", "credits" or "license" for more information.
>>> 'hoi\n'
'hoi\n'
>>> 'hoi\n'.rstrip()
'hoi'
```

Comment (r39132) [reply airflow/utils/log/logging_mixin.py]: Got it.

Comment (r39132) [reply airflow/utils/log/logging_mixin.py]: Sounds reasonable. I assuming the log call at that line will introduce a single `\n` after rstrip() removes all of the trailing white space in the buffer. @ckljohn Can you make this change? Then we should be good to merge it.

**Review by r39132 — CHANGES_REQUESTED** [review]
Sounds reasonable. I assuming the log call at that line will introduce a single \n after rstrip() removes all of the trailing white space in the buffer. @ckljohn Can you make this change? Then we should be good to merge it.

**Review by Fokko — APPROVED** [review]
Thanks @ckljohn

**Pull request merged**